modules/thermo-cycler/thermo-cycler-arduino/thermo-cycler-arduino.ino

```diff
@@ -846,18 +1096,81 @@ void therm_pid_peltier_update()
         break;
       case 4:
         temp_probes.update(ThermistorPair::cover_n_heatsink);
+        current_heatsink_temp = temp_probes.heat_sink_temperature();
         current_temperature_cover = temp_probes.cover_temperature();
+        temp_probes.plate_temp_offset = thermistor_offset();
         break;
     }
     current_temperature_plate = temp_probes.average_plate_temperature();
     timer_interrupted = false;
   }
 }
 
+bool crossed_true_target()
+{
+  if (this_step_target_temp < target_temperature_plate && current_temperature_plate >= this_step_target_temp)
+  {
+    return true;
+  }
+  else if (this_step_target_temp > target_temperature_plate && current_temperature_plate <= this_step_target_temp)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+double plate_overshoot()
+{
+  if (this_step_target_temp >= current_temperature_plate)
+  {
+    return (0.0105 * current_volume + 1.0869);
```

Comment:
where do these numbers come from? should they be in eeprom? should they be `#define`s?